pkg/ensurance/analyzer/sort/evict-sort.go

```diff
@@ -0,0 +1,182 @@
+package sort
+
+import (
+	"sort"
+
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/gocrane/crane/pkg/ensurance/executor"
+	"github.com/gocrane/crane/pkg/utils"
+)
+
+// EvictRankFunc sorts the pods in eviction order
+type EvictRankFunc func(pods []executor.EvictPod)
+
+var evictSortFunc = map[string]func(p1, p2 executor.EvictPod) int{
+	"ExtResourceEnable": evictUseExtResource,
+	"ClassAndPriority":  evictClassAndPriority,
+	"ExtResourceUsage":  evictExtCpuUsed2Limit,
+	"CpuUsage":          evictCpuUsage,
+	"RunningTime":       evictStartTime,
+}
+
+func EvictRankFuncConstruct(customize []string) EvictRankFunc {
+	var evictRankFunc EvictRankFunc
+	if len(customize) == 0 {
+		cmp := []cmpEvictFunc{}
+		for _, sortFunc := range customize {
+			if f, ok := evictSortFunc[sortFunc]; ok {
+				cmp = append(cmp,f)
+			}
+			evictRankFunc = evictOrderedBy(cmp...).Sort
+			return evictRankFunc
+		}
+	}
+	evictRankFunc = evictRankCpuPressure
+
+	return evictRankFunc
+}
+
+func evictRankCpuPressure(pods []executor.EvictPod) {
+	evictOrderedBy(evictUseExtResource, evictClassAndPriority, evictExtCpuUsed2Limit, evictCpuUsage, evictStartTime).Sort(pods)
```

Comment:
How about if user adjust the order of sort method, as it fit the expect of the user? such as user define (evictCpuUsage,evictExtCpuUsed2Limit)

Reply:
User can use EvictRankFuncConstruct from input options, but we do not need to do that now, the func can be kept in.